### ISSUE TITLE
Template generation fails for complex bounds on class types

### ISSUE THREAD

**Issue by dweiss** [carrotsearch/hppc#30]
```
public class KTypeFoo<KType extends Comparable<? extends KType>> {}
```
doesn't work. It should (even if we don't use it).

**Comment by catap** [comment]
@dweiss any plan to fix it?

**Comment by dweiss** [comment]
Nah, not really... Things do work for the simple types we use and making those templates work for anything would require a lot of effort (I think switching to a full type parser would be required). Realistically, I don't think I'll have the time to work on this in any near future, sorry!

**Comment by catap** [comment]
@dweiss maybe you now any similar template processor? I've investigated `fastutil`'s one and it... let say not reusable outside of that project :)

**Comment by dweiss** [comment]
Nope, I don't know if there's anything "generic" like this. fastutil has one that suits its purposes... and so does HPPC. A general Java template preprocessor would be an interesting project on its own but I haven't heard of anything like it.

**Comment by hiSandog** [comment]
Given the maintainer note, this probably should be treated as an explicit unsupported grammar boundary rather than an accidental template bug. A small parser diagnostic such as “wildcard bounds in template type parameters are unsupported” would be better than failing later during generation. That would also help anyone evaluating whether HPPC's template processor is suitable outside HPPC's own simple type patterns.

**Comment by dweiss** [comment]
This template processor isn't really meant to be used anywhere "outside" of the project - it is rather focused on what was needed. I'm sure better solutions could be now written. I don't think I'll be returning this this.
